ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java

```diff
@@ -440,7 +440,7 @@ long getLatestInstalledSnapshotIndex() {
    */
   long getSnapshotIndex() {
     final SnapshotInfo s = getLatestSnapshot();
-    final long latestSnapshotIndex = s != null ? s.getIndex() : 0;
+    final long latestSnapshotIndex = s != null ? s.getIndex() : -1;
```

Comment:
Let's use `RaftLog.INVALID_LOG_INDEX` instead of -1.
```suggestion
    final long latestSnapshotIndex = s != null ? s.getIndex() : RaftLog.INVALID_LOG_INDEX;
```